File: windows/tests/Elevate.Core.Tests/AdminPortalTests.cs
using Elevate.Core.Models;
using FluentAssertions;

namespace Elevate.Core.Tests;

/// <summary>The admin portals the tenant menu can open directly in a tenant.</summary>
public class AdminPortalTests
{
    private const string Tenant = "11111111-2222-3333-4444-555555555555";

    [Fact]
    public void IbizaPortalsTakeTheTenantInThePath()
    {
        AdminPortal.Azure.Uri(Tenant).AbsoluteUri.Should().Be($"https://portal.azure.com/{Tenant}");
        AdminPortal.Entra.Uri(Tenant).AbsoluteUri.Should().Be($"https://entra.microsoft.com/{Tenant}");
        AdminPortal.Intune.Uri(Tenant).AbsoluteUri.Should().Be($"https://intune.microsoft.com/{Tenant}");
    }

    [Fact]
    public void SecurityPortalsTakeTheTenantAsQuery()
    {
        AdminPortal.Defender.Uri(Tenant).AbsoluteUri.Should().Be($"https://security.microsoft.com/?tid={Tenant}");
        AdminPortal.Purview.Uri(Tenant).AbsoluteUri.Should().Be($"https://purview.microsoft.com/?tid={Tenant}");
    }

    [Fact]
    public void TenantIdIsEscaped()
    {
        AdminPortal.Azure.Uri("a b/c").AbsoluteUri.Should().Be("https://portal.azure.com/a%20b%2Fc");
        AdminPortal.Defender.Uri("a&b").AbsoluteUri.Should().Be("https://security.microsoft.com/?tid=a%26b");
    }

    [Fact]
    public void MenuOrderAndTitles()
    {
        AdminPortal.All.Should().Equal(AdminPortal.Azure, AdminPortal.Entra, AdminPortal.Intune, AdminPortal.Defender, AdminPortal.Purview);
        AdminPortal.All.Select(p => p.Title).Should().Equal("Azure Portal", "Entra admin center", "Intune admin center", "Defender Portal", "Purview Portal");
    }

    [Fact]
    public void RejectsBlankTenant()
    {
        var act = () => AdminPortal.Azure.Uri(" ");
        act.Should().Throw<ArgumentException>();
    }
}

File: macos/Tests/ElevateCoreTests/AdminPortalTests.swift
import Testing
import Foundation
@testable import ElevateCore

@Suite struct AdminPortalTests {
    let tenant = "11111111-2222-3333-4444-555555555555"

    @Test func ibizaPortalsTakeTheTenantInThePath() {
        #expect(AdminPortal.azure.url(tenantId: tenant).absoluteString == "https://portal.azure.com/\(tenant)")
        #expect(AdminPortal.entra.url(tenantId: tenant).absoluteString == "https://entra.microsoft.com/\(tenant)")
        #expect(AdminPortal.intune.url(tenantId: tenant).absoluteString == "https://intune.microsoft.com/\(tenant)")
    }

    @Test func securityPortalsTakeTheTenantAsQuery() {
        #expect(AdminPortal.defender.url(tenantId: tenant).absoluteString == "https://security.microsoft.com/?tid=\(tenant)")
        #expect(AdminPortal.purview.url(tenantId: tenant).absoluteString == "https://purview.microsoft.com/?tid=\(tenant)")
    }

    @Test func tenantIdIsEscaped() {
        #expect(AdminPortal.azure.url(tenantId: "a b/c").absoluteString == "https://portal.azure.com/a%20b%2Fc")
        #expect(AdminPortal.defender.url(tenantId: "a&b").absoluteString == "https://security.microsoft.com/?tid=a%26b")
    }

    @Test func menuOrderAndTitles() {
        #expect(AdminPortal.allCases == [.azure, .entra, .intune, .defender, .purview])
        #expect(AdminPortal.allCases.map(\.title) == ["Azure Portal", "Entra admin center", "Intune admin center", "Defender Portal", "Purview Portal"])
    }
}
